monitoring/README.md

```diff
@@ -0,0 +1,185 @@
+
+Creating a Span it requires a Tracer. The library provides a simple way to create Spans. Furthermore, exposing the Span interface you can easily add Attributes, Events, Links, Status and Record Errors in your Spans.
+
+On top of the Span interface from Otel, the library also exposes two more methods on the Span, `EndWithError` and `EndSuccessfully`.
+
+By default the Span interface from Otel exposes only the method `End` to end the Span.
+The new method `EndWithError` is also ending the Span, but it will flag the Span as errored and will also include the error into the Span Events (as it must be based on Otel). On the other hand, `EndSuccessfully` ends the Span and updates the Status as `Ok`.
+
+> [!IMPORTANT]
+> You must **always** close the Spans, otherwise you might experience OOM-kills in your services. One way to always ensure the spans are closing, you can use the [go-spancheck](https://github.com/jjti/go-spancheck) rule in the [golangci-lint](https://github.com/golangci/golangci-lint).
+
+### Automatic Correlation
+
+In order for the library to give a better experience when working with Traces, Spans and Logs, it also provides some automations to make the usage easier.
+
+When you create a new Span, it returns back a new `context.Context` value. This new context contains the Span information.
+
+When logging inside a block of code that is wrapped in a Span, it is recommended to pass to the Logger the new context that includes the Span information. That won't only inluce the Trace and Span IDs into the log, but also will automatically include into the Spans the attributes that are being passed to the Logger.
+
+That ensures useful information for debugging will be present at the same time in both the logs and the spans.
+
+> [!IMPORTANT]
+> Debug logs do not contain the Trace and Span ID and also do not attach any given metadata to the Span. That is useful because including the debug logs to the spans can flood them with unnecessary information, making them harder to interpret.
+
+> [!WARNING]
+> When you add an error log, the span will be flaged as errored and will also include the error into the Span Events (as it must be based on Otel).
+
+## Distributed Tracing
+
+**Distributed Tracing** provides visibility into the flow of requests as they traverse through various services in a distributed system. This capability is essential for understanding the interactions between microservices, identifying performance bottlenecks, and diagnosing issues. Each trace captures the entire lifecycle of a request, recording its journey across different services and components. By analyzing traces, you can visualize how requests propagate through your system and identify areas for optimization. This library supports distributed tracing by leveraging OpenTelemetry, ensuring consistent trace propagation and context management across different services and frameworks.
```

Comment:
I would move this to the `## Traces` section above and remove this `## Distributed Tracing` section. Tracing without the distributed part is a bit strange to do and it might cause confusion with some users.